### ISSUE TITLE
No power graph when imported from Bosch eBike Connect .tcx

### ISSUE THREAD

**Issue by phillipp** [tumic0/GPXSee#308]
Hi,
I was having a look at GPXSee when I noticed that it doesn't import the power graph from a .tcx file. A sample file (condensed) looks like this:

```
<?xml version="1.0" encoding="UTF-8"?><TrainingCenterDatabase xmlns="http://www.garmin.com/xmlschemas/TrainingCenterDatabase/v2" xmlns:ns2="http://www.garmin.com/xmlschemas/ActivityExtension/v2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.garmin.com/xmlschemas/TrainingCenterDatabase/v2 http://www.garmin.com/xmlschemas/TrainingCenterDatabasev2.xsd">
  <Activities>
    <Activity Sport="Biking">
      <Id>2020-09-14T10:02:16Z</Id>
      <Lap StartTime="2020-09-14T10:02:16Z">
        <TotalTimeSeconds>2218.0</TotalTimeSeconds>
        <DistanceMeters>15529.0</DistanceMeters>
        <MaximumSpeed>11.95</MaximumSpeed>
        <Calories>340</Calories>
        <AverageHeartRateBpm xsi:type="HeartRateInBeatsPerMinute_t">
          <Value>155.0</Value>
        </AverageHeartRateBpm>
        <MaximumHeartRateBpm xsi:type="HeartRateInBeatsPerMinute_t">
          <Value>195</Value>
        </MaximumHeartRateBpm>
        <Cadence>85.0</Cadence>
        <Track>
          <Trackpoint>
            <Time>2020-09-14T10:02:17Z</Time>
            <DistanceMeters>0</DistanceMeters>
            <HeartRateBpm xsi:type="HeartRateInBeatsPerMinute_t">
              <Value>95</Value>
            </HeartRateBpm>
            <Cadence>82</Cadence>
            <Extensions>
              <TPX xmlns="http://www.garmin.com/xmlschemas/ActivityExtension/v2">
                <Speed>5.611111</Speed>
                <Watts>262</Watts>
              </TPX>
            </Extensions>
          </Trackpoint>
        </Track>
      </Lap>
      <Creator xsi:type="Device_t">
        <Name>Bosch eBike Connect with barometer</Name>
        <Version>
          <BuildMajor>0</BuildMajor>
          <BuildMinor>0</BuildMinor>
        </Version>
      </Creator>
    </Activity>
  </Activities>
</TrainingCenterDatabase>

```

Looking at the `tcxparser.cpp` there is one extra element, the wrapping <TPX>, that seems not to be handled. If I remove the wrapping element with searh+replace, the power graph shows up.

Is this an exotic standard interpretation from Bosch's side or just a bug?

**Comment by gresbtim** [comment]
No heartrate = Clinical Death. Please consult with your doctor. :rip:

**Comment by tumic0** [comment]
Fixed in 0e026d6a962cf70ab6254825373d1dddc7e7f288. But note that as a GPS log file viewer, GPXSee does not display files without GPS info like the one in the example. For "regular" TCX files with coordinates, the power graph shall be now displayed.

**Comment by phillipp** [comment]
@tumic0 Thanks for your work!
